include/glow/Backends/CompiledFunction.h

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_BACKENDS_COMPILEDFUNCTION_H
+#define GLOW_BACKENDS_COMPILEDFUNCTION_H
+
+namespace glow {
+
+/// Interface containing state necessary to execute a compiled network.
+class CompiledFunction {
+public:
+  /// Dtor.
+  virtual ~CompiledFunction() = default;
+
+  /// Execute the network.
+  virtual void doForwardPass() = 0;
```

Comment:
We actually do not have a separate backward pass since a while, because we just generate it after the forward pass into the same IRFunction. And we use `doForwardPass` even for training these days. We should probably rename this method into `run` or `execute` or something like that. But this could be done here and elsewhere in a separate PR.